float64 distance
float64 new_theta
float64 initial_theta
float64 x_target
float64 y_target